### PR TITLE
Feature/initial implementation

### PR DESCRIPTION
**Pull request by markysoft**
Initial work on containerising the profiles-db data update, including scheduling and retreiving ETL json files

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

File: lib/mongodb/dataImporter.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  async importData(data) {
+    try {
+      await this.mongoWrapper.openConnection();
+      await this.cleanupTempCollection();
+      await this.insertDataToTemporaryCollection(data);
+      await this.validateData();
+      await this.removeOriginalCollection();
+      await this.renameTempCollection();
```
Comment (neilbmclaughlin) [inline lib/mongodb/dataImporter.js]: Interestingly, ElasticSearch uses 'aliases' to allow rolling updates which seems useful. It is an outstanding feature request in MongoDB  (https://jira.mongodb.org/browse/SERVER-3498)

File: test/integration/merge.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    'EMIS',
+    'INPS',
+    'Informatica',
+    'Microtest',
+    'NK',
+    'TPP'
+  ];
+
+  if (supplier.indexOf('(I)') > 0) {
+    expect(supplier).to.have.string('(I)');
```
Comment (neilbmclaughlin) [inline test/integration/merge.js]: Not sure I understand what is being tested here.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,192 @@
...
+All other members will be present and populated.
+
+## Interrogating the json with [jq](https://stedolan.github.io/jq/)
+
+* List suppliers: `jq -c 'unique_by(.Supplier) | [.[].Supplier]' input/pomi.json`
+* Find single item by `odsCode`:
+  `jq '.[] | select(.odsCode == "${odsCode}")' data/gp-data-merged.json`
+
+## Environment variables
+
```
Comment (markysoft) [inline README.md]: On other projects the snyk tests don't run locally, only in travis. I'll change this project to behave the same

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  profiles-db-updater:
+    extends:
+      service: base-app
+    links:
+    - mongo-db:mongo
+
+  mongo-db:
+    image: mongo:3.4
+    ports:
```
Comment (markysoft) [inline docker-compose.yml]: this file is only used locally, the one in the rancher-config folder is the production one which has not mongodb in it.

File: test/integration/merge.js (hunk 2 above)
Comment (markysoft) [inline test/integration/merge.js]: It came from the db-profiles tests, I think it's saying a supplier is valid if it has `(I)` in the name, I'll change it to make it clearer.

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

File: test/integration/merge.js [hunk 5]
```diff
@@ -15,9 +15,8 @@ function expectSupplierToBeValid(supplier) {
     'TPP'
   ];
 
-  if (supplier.indexOf('(I)') > 0) {
-    expect(supplier).to.have.string('(I)');
-  } else {
+  // valid supplier is either in the list or contains (I)
+  if (supplier.indexOf('(I)') < 0) {
```
Comment (neilbmclaughlin) [inline test/integration/merge.js]: Does this mean that "Blah(I)" is valid input?

It's a bit of a pain but would it be better to explicitly add the "*(I)" variants to the valid suppliers? I'm always wary about conditionals expectations in tests.

Comment (markysoft) [inline test/integration/merge.js]: Sure will do

**Comment by c2s-dev** [comment]
:rocket: deployment of profiles-db-updater succeeded (http://profiles-db-updater-pr-1.dev.beta.nhschoices.net)

**Pull request merged**